### PR TITLE
[Keyboard] Add cf81 keyboard.

### PR DESCRIPTION
**Pull request by jonylee1986**
## Description
As title.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/chosfox/cf81/cf81.c [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+        case DF(WIN_B):
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(WIN_B);
+                layer_state_set(1<<WIN_B);
```
Comment (drashna) [inline keyboards/chosfox/cf81/cf81.c:176]: why set the layer here, when setting the default layer?  That's kind of at odds with the behavior of the two layer masks. 

Also, if you want to set the default layer to be persistent from keycode, you could do this in `default_layer_state_set_kb`: 
https://github.com/qmk/qmk_firmware/blob/26227d1f66673a5c16bef72971202811cd4b5eae/users/drashna/callbacks.c#L190-L202

File: keyboards/chosfox/cf81/cf81.c [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            }
+            return false;
+        case DF(MAC_B):
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(MAC_B);
+                layer_state_set(1<<MAC_B);
+                keymap_config.no_gui     = 0;
```
Comment (drashna) [inline keyboards/chosfox/cf81/cf81.c:183]: Any specific reason for disabling win/gui/cmd on the mac layer?  The key is still usable and used heavily

File: keyboards/chosfox/cf81/config.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* Use 6 dynamic keymap layers */
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (drashna) [inline keyboards/chosfox/cf81/config.h:20]: 6 or 4?

**Review by drashna — APPROVED** [review]

File: keyboards/chosfox/cf81/rules.mk [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Build Options
+#   change yes to no to disable
+#
+WEAR_LEVELING_DRIVER = spi_flash
+
```
Comment (waffle87) [inline keyboards/chosfox/cf81/rules.mk:5]:
```suggestion
WEAR_LEVELING_DRIVER = spi_flash
```

**Review by waffle87 — APPROVED** [review]

**Pull request merged**